pandas/tseries/timedeltas.py

```diff
@@ -142,13 +137,50 @@ def _validate_timedelta_unit(arg):
 
 
 def _coerce_scalar_to_timedelta_type(r, unit='ns', box=True, errors='raise'):
-    """
-    convert strings to timedelta; coerce to Timedelta (if box), else
-    np.timedelta64
-    """
+    """Convert string 'r' to a timedelta object."""
+
+    try:
+        result = tslib.convert_to_timedelta64(r, unit)
```

Comment:
``tslib.convert_to_timedelta64`` *already* accepts the coerce arg, you are doing extra unecessary things here